### PR TITLE
Add more help documentation, automatically generate CLI reference

### PR DESCRIPTION
**Pull request by cdrage**
- Adds more help documentation
- Adds script to generate CLI reference

### CODE REVIEW

**Comment by cdrage** [comment]
Still need to add a lot more documentation. But this is my progress so far.

**Comment by cdrage** [comment]
Ready for review @kadel 

Mainly, look at the structure here: https://github.com/cdrage/odo/blob/6710e871e39054b5aa607f55e234de78a969db4e/docs/cli-reference.md

All the other "fluff" in the code is simple edits to `cmd` help menus. I've also added a script to automatically generate this on each-release.

**Comment by cdrage** [comment]
@kadel Updated. Ready for another review / merge :+1:

File: cmd/url.go [hunk 1]
```diff
@@ -20,24 +20,27 @@ var urlCmd = &cobra.Command{
...
 
-# Create a URL for a specific component
-odo url create <component name>
+The created URL can be used to access the specified component from outside the OpenShift cluster.
 `,
+	Example: `  # Create a URL for the current component.
+  odo url create
+
+  # Create a URL for a specific component
+  odo url create myurl
```
Comment (kadel) [inline cmd/url.go]: `myurl` might be a bit misleading. The argument is a name of a component. This looks like it is a name of an url.
`mycomponent` might be better

File: cmd/url.go [hunk 2]
```diff
@@ -69,12 +72,10 @@ odo url create <component name>
...
 	Short: "Delete a URL",
-	Long: `Delete a URL.
-Deleted the given URL, hence making the service inaccessible.
-`,
-	Example: `# Delete a URL to a component
-odo url delete <URL>
-`,
+	Long:  `Delete the given URL, hence making the service inaccessible.`,
+	Example: `  # Delete a URL to a component
+  odo url delete myurl
```
Comment (kadel) [inline cmd/url.go]: here is `myurl` ok, as delete has URL name as an argument.

**Comment by cdrage** [comment]
@kadel Updated!

**Pull request merged**